lib/cylc/config.py

```diff
@@ -211,6 +211,13 @@ def __init__(self, suite, fpath, template_vars=[],
                     if item != 'graph' and value.get('graph'):
                         just_has_async_graph = False
                         break
+                icp = self.cfg['scheduling'].get('initial cycle point')
+                fcp = self.cfg['scheduling'].get('final cycle point')
+                if just_has_async_graph and not (
+                        icp in [None, "1"] and (fcp == icp or fcp is None)):
```

Comment:
I don't mean to be a pain, but I think it will be prettier to write either:

```python
icp in [None, "1"] and fcp in [None, icp]
```
or:
```python
(icp == "1" or icp is None) and (fcp == icp or fcp is None)
```